ansible_collections/arista/avd/plugins/plugin_utils/schema/avddataconverter.py

```diff
@@ -1,238 +1,166 @@
+            for resolved_key in resolved_keys:
+                dynamic_keys.setdefault(resolved_key, childschema)
+
+        for key, childschema in dynamic_keys.items():
+            if key not in data:
+                # Skip key since there is nothing to convert if the key is not set in data
+                continue
+
+            # Perform type conversion of the data for the child key if required based on "convert_types"
+            if "convert_types" in childschema:
+                yield from self.convert_types(childschema["convert_types"], data, key, childschema, path + [key])
+
+            # Dive in to child keys/schemas
+            yield from self.convert_data(data[key], childschema, path + [key])
+
+    def convert_items(self, items: dict, data: list, schema: dict, path: list):
+        """
+        This function performs conversion on each item with the items subschema
+        """
+        if not isinstance(data, list):
+            return
+
+        for index, item in enumerate(data):
+            # Perform type conversion of the items data if required based on "convert_types"
+            if "convert_types" in items:
+                yield from self.convert_types(items["convert_types"], data, index, items, path + [f"[{index}]"])
+
+            # Dive in to child items/schema
+            yield from self.convert_data(item, items, path + [f"[{index}]"])
+
+    def convert_types(self, convert_types: list, data: dict | list, index: str | int, schema: dict, path: list):
+        """
+        This function performs type conversion if necessary on a single data instance.
+        It is invoked for child keys during "keys" conversion and for child items during
+        "items" conversion
+
+        "data" is either the parent dict or the parent list.
+        "index" is either the key of the parent dict or the index of the parent list.
+
+        Conversion is performed in-place using the provided "data" and "index"
+
+        Yields AvdConversionWarning and/or AvdDeprecationWarning except for simple str/int/bool conversions
+
+        Any conversion errors are ignored and the original value is returned
+        """
+        schema_type = schema.get("type")
+
+        # Get value from input data
+        value = data[index]
+
+        # For simple conversions, skip conversion if the value is of the correct type
+        if schema_type in SIMPLE_CONVERTERS and isinstance(value, SCHEMA_TO_PY_TYPE_MAP.get(schema_type)):
+            return
+
+        # Prepare string for var path used in warning messages.
+        path_str = ".".join(path)
+
+        for convert_type in convert_types:
+            if isinstance(value, SCHEMA_TO_PY_TYPE_MAP.get(convert_type)):
+                if schema_type in SIMPLE_CONVERTERS:
+                    try:
+                        data[index] = SIMPLE_CONVERTERS[schema_type](value)
+                    except Exception:
+                        # Ignore errors
```

Comment:
probably worth adding a log message at debug level if this happens

Reply:
Added TODOs since this code will be refactored (again) in the deprecation PR. We could add another error type we can yield for this case.